cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -0,0 +1,163 @@
+
+    cron_job = client.V1beta1CronJob(
+        api_version="batch/v1beta1",
+        kind="CronJob",
+        metadata=client.V1ObjectMeta(
+            name=f'{pod_name}-cronjob'
+        ), spec=client.V1beta1CronJobSpec(
+            schedule=schedule,
+            job_template=client.V1beta1JobTemplateSpec(
+                spec=client.V1JobSpec(
+                    template=client.V1PodTemplateSpec(
+                        spec=_get_pod_manifest(
+                            pod_name, image, env_variables, secrets, command, args
+                        )
+                    )
+                )
+            ),
+        )
+    )
+    try:
+        api_response = client.BatchV1beta1Api().create_namespaced_cron_job(
+            namespace="default",
+            body=cron_job,
+            pretty='true',
+            dry_run='All' if dry_run else None
+        )
+        logging.info(f"CronJob created: {api_response}")
+    except rest.ApiException as e:
+        logging.error(f"Error creating CronJob: {e}")
+        raise
+    return pod_name
+
+
+def _get_pod_manifest(
+        pod_name: Text,
+        image: Text,
+        env_variables: List[Dict[Text, Text]],
+        secrets: List[Dict[Text, Text]],
+        command: List[Text],
+        args: List[Text],
+):
+    return toolz.pipe(
+        (pod_name, image),
+        gamla.star(_make_base_pod_spec),
+        gamla.assoc_in(keys=["containers", 0, "env"], value=env_variables),
+        toolz.compose_left(
+            toolz.juxt(*map(_add_volume_from_secret, secrets)), toolz.merge
+        )
+        if secrets
+        else toolz.identity,
+        gamla.assoc_in(keys=["containers", 0, "command"], value=command)
+        if command
+        else toolz.identity,
+        gamla.assoc_in(keys=["containers", 0, "args"], value=args)
+        if args
+        else toolz.identity,
+    )
+
+
+def _make_base_pod_spec(pod_name: Text, image: Text):
+    return {
+        "containers": [{"image": image, "name": f"{pod_name}-container"}],
+        "imagePullSecrets": [
+            {"name": f"{_get_repo_name_from_image(image)}-gitlab-creds"}
+        ],
+        "restartPolicy": "Never",
+    }
+
+
+def _add_volume_from_secret(secret: Dict[Text, Text]):
+    return toolz.compose_left(
+        gamla.assoc_in(
+            keys=["containers", 0, "volumeMounts"],
+            value=[
+                {
+                    "name": secret["volume_name"],
+                    "mountPath": secret["mount_path"],
+                    "readOnly": True,
+                }
+            ],
+        ),
+        gamla.assoc_in(
+            keys=["volumes"],
+            value=[
+                {
+                    "name": secret["volume_name"],
+                    "secret": {"secretName": secret["secret_name"]},
+                }
+            ],
+        ),
+    )
+
+
+make_deploy_cron_job = gamla.compose_left(
+    _get_kubernetes_client,
+    toolz.juxt(_create_cron_job),
```

Comment:
looks like the juxt is redundant, you have only one function, also the compose left below